main/pipeline/pipeline.cc

```diff
@@ -983,9 +983,11 @@ ast::ParsedFilesOrCancelled typecheck(unique_ptr<core::GlobalState> &gs, vector<
                                                            file.data(igs).path());
                                     }
                                     // Stream out errors
-                                    threadResult.counters = getAndClearThreadCounters();
-                                    resultq->push(move(threadResult), processedByThread);
-                                    processedByThread = 0;
+                                    if (processedByThread >= 10) {
+                                        threadResult.counters = getAndClearThreadCounters();
```

Comment:
Can we do counters only once, in the final result?

Reply:
That makes sense to me.

Reply:
Actually, I think we might need to do a second round of worker work to aggregate counters.

If `processedByThread` is 0 at the end of the loop, then the coordinator thread may have already moved on from typecheck and won't get a subsequent message.